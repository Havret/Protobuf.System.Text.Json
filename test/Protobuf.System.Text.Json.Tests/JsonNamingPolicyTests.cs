using System.Text.Json;
using System.Text.Json.Protobuf.Tests;
using Protobuf.System.Text.Json.Tests.Utils;
using SmartAnalyzers.ApprovalTestsExtensions;
using Xunit;

namespace Protobuf.System.Text.Json.Tests;

public class JsonNamingPolicyTests
{
    [Fact]
    public void Should_serialize_message_with_primitive_types()
    {
        // Arrange
        var msg = new SimpleMessage
        {
            DoubleProperty = 2.5d
        };
        var jsonSerializerOptions = TestHelper.CreateJsonSerializerOptions();
        jsonSerializerOptions.PropertyNamingPolicy = new JsonLowerCaseNamingPolicy();

        // Act
        var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);

        // Assert
        var approver = new ExplicitApprover();
        approver.VerifyJson(serialized);
    }
    
    [Fact]
    public void Should_ignore_PropertyNamingPolicy_when_UseProtobufJsonNames_set_to_true()
    {
        // Arrange
        var msg = new SimpleMessage
        {
            DoubleProperty = 2.5d
        };
        var jsonSerializerOptions = new JsonSerializerOptions();
        jsonSerializerOptions.PropertyNamingPolicy = new JsonLowerCaseNamingPolicy();
        jsonSerializerOptions.AddProtobufSupport(options => options.UseProtobufJsonNames = true);
        
        // Act
        var serialized = JsonSerializer.Serialize(msg, jsonSerializerOptions);

        // Assert
        var approver = new ExplicitApprover();
        approver.VerifyJson(serialized);
    }

    private class JsonLowerCaseNamingPolicy : JsonNamingPolicy
    {
        public override string ConvertName(string name)
        {
            return name.ToLower();
        }
    }
}