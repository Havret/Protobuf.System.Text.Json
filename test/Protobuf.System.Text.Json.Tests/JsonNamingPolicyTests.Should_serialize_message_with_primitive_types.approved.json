{
  "doubleproperty": 2.5,
  "floatproperty": 0,
  "int32property": 0,
  "int64property": 0,
  "uint32property": 0,
  "uint64property": 0,
  "sint32property": 0,
  "sint64property": 0,
  "fixed32property": 0,
  "fixed64property": 0,
  "sfixed32property": 0,
  "sfixed64property": 0,
  "boolproperty": false,
  "stringproperty": "",
  "bytesproperty": ""
}